sa/database.go

```diff
@@ -76,6 +68,50 @@ func NewDbMap(driver string, dbConnect string) (*gorp.DbMap, error) {
 	return dbmap, err
 }
 
+// recombineURLForDB transforms a database URL to a URL-like string
+// that the mysql driver can use. The mysql driver needs the Host data
+// to be wrapped in "tcp()" but url.Parse will escape the parentheses
+// and the mysql driver doesn't understand them. So, we can't have
+// "tcp()" in the configs, but can't leave it out before passing it to
+// the mysql driver. Similarly, the driver needs the password and
+// username unescaped. Compromise by doing the leg work if the config
+// says the database URL's scheme is a fake one called
+// "mysqltcp://". See
```

Comment:
There are other params we'll want to enforce like strict and charset.

I'm strongly opposed to using regexs for validation and, while I'm not willing to write the code, I would be glad for this to wait for someone else to write it and for this to be rebased on top of it.